subprojects/language-native/src/main/java/org/gradle/language/nativeplatform/tasks/AbstractNativeCompileTask.java

```diff
@@ -243,6 +247,25 @@ public void includes(Object includeRoots) {
         includes.from(includeRoots);
     }
 
+    /**
+     * Returns the system include directories to be used for compilation.
+     *
+     * @since 4.8
+     */
+    @Internal("The paths for include directories are tracked via the includePaths property, the contents are tracked via discovered inputs")
+    public ConfigurableFileCollection getSystemIncludes() {
+        return systemIncludes;
+    }
+
+    /**
+     * Add system include directories where the compiler should search for header files.
+     *
+     * @since 4.8
+     */
+    public void systemIncludes(Object systemIncludeRoots) {
```

Comment:
I think we can limit the API exposure to just `getSystemIncludes()`.  I think that makes it more "Provider" like.

Reply:
Fair enough.  Removed.